pkg/apis/aws/v1alpha1/shared_types.go

```diff
@@ -1,56 +1,138 @@
-	},
-	"ca-central-1": {
-		"initializationAMI": "ami-06ca3c0058d0275b3",
-	},
-	"eu-central-1": {
-		"initializationAMI": "ami-09de4a4c670389e4b",
-	},
-	"eu-west-1": {
-		"initializationAMI": "ami-0202869bdd0fc8c75",
-	},
-	"eu-west-2": {
-		"initializationAMI": "ami-0188c0c5eddd2d032",
-	},
-	"eu-west-3": {
-		"initializationAMI": "ami-0c4224e392ec4e440",
-	},
-	"ap-northeast-1": {
-		"initializationAMI": "ami-00b95502a4d51a07e",
-	},
-	"ap-northeast-2": {
-		"initializationAMI": "ami-041b16ca28f036753",
-	},
-	"ap-south-1": {
-		"initializationAMI": "ami-0963937a03c01ecd4",
-	},
-	"ap-southeast-1": {
-		"initializationAMI": "ami-055c55112e25b1f1f",
-	},
-	"ap-southeast-2": {
-		"initializationAMI": "ami-036b423b657376f5b",
-	},
-	"sa-east-1": {
-		"initializationAMI": "ami-05c1c16cac05a7c0b",
-	},
+import (
+	"errors"
+
+	"github.com/openshift/aws-account-operator/pkg/apis/aws/v1alpha1"
+	"github.com/openshift/aws-account-operator/pkg/awsclient"
+)
+
+const (
+	AWSRegionDefaultGlobal = "us-east-1"
+	AWSRegionDefaultChina  = "cn-north-1"
+
+	AWSSecretNameGlobal = "aws-account-operator-credentials"
+	AWSSecretNameChina  = "aws-account-operator-china-credentials"
+
+	AWSARNPrefixGlobal = "arn:aws:"
+	AWSARNPrefixChina  = "arn:aws-cn:"
+
+	AWSIAMPolicyAdministrator = "iam::aws:policy/AdministratorAccess"
+
+	AWSFedEndpointURLGlobal = "https://signin.aws.amazon.com/federation"
+	AWSFedEndpointURLChina  = "https://signin.amazonaws.cn/federation"
+
+	AWSFedConsURLGlobal = "https://console.aws.amazon.com/"
+	AWSFedConsURLChina  = "https://console.amazonaws.cn/"
+)
+
+// AwsPlatformConfig contains all required fields to service either AWS Global or AWS China.
+type AwsPlatformConfig struct {
+	ClientInput      awsclient.NewAwsClientInput
+	ARNPrefix        string
+	CoveredRegions   map[string]map[string]string
+	FederationConfig AwsFederationConfig
+}
+
+// AwsFederationConfig consolidates EndpointURL and ConsoleURL into one structure that accounts
+// for requirements of FederationConfig.
+type AwsFederationConfig struct {
+	EndpointURL string
+	ConsoleURL  string
 }
 
+var (
+	// AwsPlatformConfigGlobal provides a default struct with fields configured for AWS Global.
+	AwsPlatformConfigGlobal = AwsPlatformConfig{
+		ClientInput: awsclient.NewAwsClientInput{
+			AwsRegion:  AWSRegionDefaultGlobal,
+			SecretName: AWSSecretNameGlobal,
+			NameSpace:  v1alpha1.AccountCrNamespace,
+		},
+		ARNPrefix:      AWSARNPrefixGlobal,
+		CoveredRegions: AWSRegionsGlobal,
+		FederationConfig: AwsFederationConfig{
+			EndpointURL: AWSFedEndpointURLGlobal,
+			ConsoleURL:  AWSFedConsURLGlobal,
+		},
+	}
+
+	// AwsPlatformConfigChina provides a default struct with fields configured for AWS China.
+	AwsPlatformConfigChina = AwsPlatformConfig{
+		ClientInput: awsclient.NewAwsClientInput{
+			AwsRegion:  AWSRegionDefaultChina,
+			SecretName: AWSSecretNameChina,
+			NameSpace:  v1alpha1.AccountCrNamespace,
+		},
+		ARNPrefix:      AWSARNPrefixChina,
```

Comment:
Since we're proposing an alternative to the approach of having different `AwsPlatformConfig` objects, probably it would be better having the ARNs coming from a method called from `awsclient` interface. 
Since in the awsclient has the region, we could have a `GetARNPrefix()` method

Reply:
This would require more changes to pass around the region. I recommend doing this in a separate PR for refactoring.